post-types/ckan-backend-local-harvester.php

```diff
@@ -342,4 +341,147 @@ public static function get_source_type_form_field_options() {
 
 		return $source_type_options;
 	}
+
+	/**
+	 * Applies harvest-status filter
+	 *
+	 * @param WP_Query $query The current query.
+	 */
+	public function filter_posts_by_harvest_status( $query ) {
+		global $post_type, $pagenow;
+
+		if (
+			// Only filter when were on the edit page of ckan-local-harvesters
+			self::POST_TYPE === $post_type &&
+			'edit.php' === $pagenow &&
+			// Only filter when ckan-local-harvesters are queried
+			! empty( $query->query_vars['post_type'] ) &&
+			$query->query_vars['post_type'] === self::POST_TYPE
+		) {
+			$harvest_status_filter   = '';
+			if ( isset( $_GET['harvest_status_filter'] ) ) {
+				$harvest_status_filter = sanitize_text_field( $_GET['harvest_status_filter'] );
+			}
+
+			$harvest_ids_by_state = self::get_harvest_ids_by_status();
+
+			if ( ! empty( $harvest_status_filter ) ) {
+				// wp_query does not allow an empty array of meta_values, so an empty meta_value is given instead
+				if ( empty( $harvest_ids_by_state[ $harvest_status_filter ] ) ) {
+					$harvest_ids = array( '' );
+				} else {
+					$harvest_ids = $harvest_ids_by_state[ $harvest_status_filter ];
+				}
```

Comment:
Make sure `get_harvest_ids_by_status` returns `array( '' )` if no harvesters have a certain status instead of handing this with a if/else here